common/src/bytecode.rs

```diff
@@ -207,4 +210,63 @@ mod tests {
     fn empty_hex_bytecode_is_empty() {
         assert!(Bytecode::from_hex_str("0x").unwrap().is_empty());
     }
+
+    #[test]
+    fn to_fixed_hex_() {
+        for (value, expected) in &[
+            (
+                "0x0000000000000000000000000000000000000000",
+                "0000000000000000000000000000000000000000",
+            ),
+            (
+                "0x0102030405060708091020304050607080900001",
+                "0102030405060708091020304050607080900001",
+            ),
+            (
+                "0x9fac3b52be975567103c4695a2835bba40076da1",
+                "9fac3b52be975567103c4695a2835bba40076da1",
+            ),
+        ] {
+            let value: Address = value[2..].parse().unwrap();
+            assert_eq!(to_fixed_hex(&value), *expected);
+        }
+    }
+
+    #[test]
+    fn bytecode_link_success() {
+        let address = Address::zero();
+        let address_encoded = [0u8; 20];
+        let name = "name";
+        let placeholder = format!("__{:_<38}", name);
+        let mut bytecode = Bytecode::from_hex_str(format!(
+            "0x61{}{}61{}",
+            placeholder, placeholder, placeholder
+        ))
+        .unwrap();
+        bytecode.link(name, address).unwrap();
+        let bytes = bytecode.to_bytes().unwrap();
+        let mut expected = Vec::<u8>::new();
+        expected.extend(&[0x61]);
+        expected.extend(&address_encoded);
+        expected.extend(&address_encoded);
+        expected.extend(&[0x61]);
+        expected.extend(&address_encoded);
+        assert_eq!(bytes.0, expected);
+    }
+
+    #[test]
+    fn bytecode_link_fail() {
+        let address = Address::zero();
+        let placeholder = format!("__{:_<38}", "name0");
+        let mut bytecode = Bytecode::from_hex_str(format!(
+            "0x61{}{}61{}",
+            placeholder, placeholder, placeholder
+        ))
+        .unwrap();
+        // name does not match
+        match bytecode.link("name1", address) {
+            Err(LinkError::NotFound(_)) => (),
+            _ => assert!(false, "should fail with not found error"),
```

Comment:
Its failing because of a clippy lint:
```suggestion
            _ => panic!("should fail with not found error"),
```